packages/app/src/components/TokenBalance.tsx

```diff
@@ -0,0 +1,70 @@
+'use client'
+import { useBalance, useReadContract } from 'wagmi'
+import { formatEther } from 'viem'
+import { toBigInt } from 'ethers'
+import { useEffect } from 'react'
+
+const abi = [
+  {
+    type: 'function',
+    name: 'balanceOf',
+    stateMutability: 'view',
+    inputs: [{ name: 'account', type: 'address' }],
+    outputs: [{ type: 'uint256' }],
+  },
+] as const
+
+interface TokenBalanceProps {
+  readonly address: `0x${string}`
+  readonly tokenAddress?: `0x${string}`
+  readonly className?: string
+  readonly toFixed?: number
+  readonly onBalanceChange?: ({
+    balance,
+    formattedBalance,
+  }: {
+    balance: bigint
+    formattedBalance: string | null
+  }) => void
+}
+
+const formatBalance = (balance: bigint, toFixed?: number) => {
```

Comment:
Should we move this move this to a utils function? I see its on both examples pages, but probably useful for other components. 

Can be done as separate PR. Happy to merge this first, if that's easier.